spec/html_preprocessor_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+
+describe PDFKit::HTMLPreprocessor do
```

Comment:
I would like to see another quick test for when root_url & protocol are nil. This would be the case for when PDFKit is not being used as middleware & the user doesn't explicitly set root_url & protocol. It's a simple test that prevents unintentional breakage in the future.